docs/okeps/okep-5252-multiple-primary-udns.md

```diff
@@ -0,0 +1,208 @@
+UDN by creating the pod with the annotation "k8s.ovn.org/requested-primary-udn". If the value of this
+annotation does not correspond to a legitimate primary UDN, the pod will not be wired and will error. If
+a user applies this annotation to a pod, whose namespace is using the custer default network (CDN), it will
+have no effect, and the pod will still come up and be wired to the CDN. If a user attempts to modify this
+annotation after a pod has been created, it will have no effect.
+
+### Kubernetes Features Implications
+
+#### Network Policy
+
+Network policy should continue to work between pods on multiple primary UDNs. The pods
+are still selectable via the API, and should still create rules on the OVN logical
+switches where the affected source pod lives. Note, this may result in O(n) additional
+ACLs for n networks depending on the network policy selectors.
+
+#### Services
+
+Services are usually isolated to each UDN, unless exposed externally. However, with multiple
+primary UDNs these services must be reachable via all pods within the namespace. Therefore,
+when a service is created within a namespace with multiple primary UDNs, the OVN loadbalancer
+will be placed on the logical switches for all primary UDNs affecting said namespace.
+
+For externally exposed services, ingress via br-ex will be delivered only one of the OVN gateway
+routers (GRs).
+
+#### Egress IP
+
+Egress IP is applied via routes in within OVN routers that force packets to cross the overlay
+to the egress node. At that point the router then sends the packet out into br-ex or the host,
+where SNAT rules modify the source IP to be the Egress IP and send the packet out.
+
+The same behavior will continue to work for pods that exist across multiple primary UDNs. The
+impact of multiple UDNs will be that the egress IP routes and policies must exist
+on each router for multiple UDNs.
+
+#### QoS
+
+QoS is applied locally to the OVN switch, so as with other features, QoS rules may need to be applied
+to multiple PUDN logical switches. QoS should continue to function as it does today, without limitations.
+
+### API Details
+
+There shall be no API changes, other than the new annotation "k8s.ovn.org/requested-primary-udn", which is applied
+to a pod at create time to signal which primary UDN the pod would like to connect to.
+
+### Implementation Details
+
+The ovnkube cluster manager will need to be modified to validate multiple primary UDNs according to this spec. OVNKube-controller
+will need to be modified to track multiple PUDNs per namespace, and ensure that each controller applies its feature
+configuration across multiple networks.
+
+For the OVN topology itself, a new point to point link will be created between multiple PUDNs under the same namespace(s):
+
+![Multiple Primary UDN](../images/multiple_primary_udn.png)
+
+Here there are two PUDNS, a Layer 3 and a Layer 2 that are part of the same namespace. A point to point link is created
```

Comment:
but this is interconnecting UDNs work that's already https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5246 right? what we spoke about a few days ago?